File: examples/xatBlogAPI-lib.java
import com.google.gson.Gson;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Map;

public class XatBlogAPI {
    private static final String BASE_URL = "https://api.xatblog.net/";
    private static final Gson gson = new Gson();

    private Map<String, Object> sendRequest(String path) throws IOException {
        URL url = new URL(BASE_URL + path);
        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setRequestMethod("GET");

        int responseCode = connection.getResponseCode();

        if (responseCode == HttpURLConnection.HTTP_OK) {
            BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
            StringBuilder response = new StringBuilder();
            String line;

            while ((line = reader.readLine()) != null) {
                response.append(line);
            }

            reader.close();

            return gson.fromJson(response.toString(), Map.class);
        } else {
            return Map.of("error", "Request failed with response code " + responseCode);
        }
    }

    public Map<String, Object> latest() throws IOException {
        return sendRequest("/latest");
    }

    public Map<String, Object> prices() throws IOException {
        return sendRequest("/prices");
    }

    public Map<String, Object> countdown() throws IOException {
        return sendRequest("/countdown");
    }

    public Map<String, Object> promoted() throws IOException {
        return sendRequest("/promoted");
    }

    public Map<String, Object> activePawns() throws IOException {
        return sendRequest("/activepawns");
    }

    public Map<String, Object> regToId(String username) throws IOException {
        String endpoint = "/reg2id/" + username;
        return sendRequest(endpoint);
    }

    public Map<String, Object> idToReg(String id) throws IOException {
        String endpoint = "/id2reg/" + id;
        return sendRequest(endpoint);
    }

    public Map<String, Object> namePrice(String username) throws IOException {
        String endpoint = "/nameprice/" + username;
        return sendRequest(endpoint);
    }

    public Map<String, Object> chatPrice(String chatname) throws IOException {
        String endpoint = "/chatprice/" + chatname;
        return sendRequest(endpoint);
    }

    public Map<String, Object> chatInfo(String chatname) throws IOException {
        String endpoint = "/chatinfo/" + chatname;
        return sendRequest(endpoint);
    }

    public Map<String, Object> powerSearch(String powername) throws IOException {
        String endpoint = "/powersearch/" + powername;
        return sendRequest(endpoint);
    }

    public Map<String, Object> powerInfo(String powername) throws IOException {
        String endpoint = "/powerinfo/" + powername;
        return sendRequest(endpoint);
    }

    public Map<String, Object> powerLogs(String powername) throws IOException {
        String endpoint = "/powerlogs/" + powername;
        return sendRequest(endpoint);
    }

    public Map<String, Object> promoPrice(int hours, String language) throws IOException {
        String endpoint = "/promoprice/" + hours + "/" + language;
        return sendRequest(endpoint);
    }

    public Map<String, Object> promoPrice(int hours) throws IOException {
        return promoPrice(hours, "en");
    }

    public Map<String, Object> daysToXats(int amount) throws IOException {
        String endpoint = "/dx/" + amount;
        return sendRequest(endpoint);
    }

    public Map<String, Object> xatsToDays(int amount) throws IOException {
        String endpoint = "/x2d/" + amount;
        return sendRequest(endpoint);
    }

    public Map<String, Object> verifyBanner(String url) throws IOException {
        String endpoint = "/verifybanner/" + url;
        return sendRequest(endpoint);
    }

    public Map<String, Object> userGifts(String userOrId) throws IOException {
        String endpoint = "/usergifts/" + userOrId;
        return sendRequest(endpoint);
    }

    public Map<String, Object> jinxList() throws IOException {
        return sendRequest("/jinxlist");
    }

    public Map<String, Object> jinxList(String powername) throws IOException {
        String endpoint = "/jinxlist/" + powername;
        return sendRequest(endpoint);
    }
}

File: examples/xatBlogAPI-lib.kt
import java.net.HttpURLConnection
import java.net.URL
import com.google.gson.Gson

class XatBlogAPI {
    companion object {
        const val BASE_URL = "https://api.xatblog.net/"
    }

    private fun sendRequest(path: String): Map<String, Any> {
        val url = URL("$BASE_URL$path")
        val connection = url.openConnection() as HttpURLConnection
        connection.requestMethod = "GET"
        val responseCode = connection.responseCode

        return if (responseCode == HttpURLConnection.HTTP_OK) {
            val responseString = connection.inputStream.bufferedReader().use { it.readText() }
            Gson().fromJson(responseString, Map::class.java) as Map<String, Any>
        } else {
            mapOf("error" to "Request failed with response code $responseCode")
        }
    }

    fun latest(): Map<String, Any> {
        return sendRequest("/latest")
    }

    fun prices(): Map<String, Any> {
        return sendRequest("/prices")
    }

    fun countdown(): Map<String, Any> {
        return sendRequest("/countdown")
    }

    fun promoted(): Map<String, Any> {
        return sendRequest("/promoted")
    }

    fun activePawns(): Map<String, Any> {
        return sendRequest("/activepawns")
    }

    fun regToId(username: String): Map<String, Any> {
        val endpoint = "/reg2id/$username"
        return sendRequest(endpoint)
    }

    fun idToReg(id: String): Map<String, Any> {
        val endpoint = "/id2reg/$id"
        return sendRequest(endpoint)
    }

    fun namePrice(username: String): Map<String, Any> {
        val endpoint = "/nameprice/$username"
        return sendRequest(endpoint)
    }

    fun chatPrice(chatname: String): Map<String, Any> {
        val endpoint = "/chatprice/$chatname"
        return sendRequest(endpoint)
    }

    fun chatInfo(chatname: String): Map<String, Any> {
        val endpoint = "/chatinfo/$chatname"
        return sendRequest(endpoint)
    }

    fun powerSearch(powername: String): Map<String, Any> {
        val endpoint = "/powersearch/$powername"
        return sendRequest(endpoint)
    }

    fun powerInfo(powername: String): Map<String, Any> {
        val endpoint = "/powerinfo/$powername"
        return sendRequest(endpoint)
    }

    fun powerLogs(powername: String): Map<String, Any> {
        val endpoint = "/powerlogs/$powername"
        return sendRequest(endpoint)
    }

    fun promoPrice(hours: Int, language: String = "en"): Map<String, Any> {
        val endpoint = "/promoprice/$hours/$language"
        return sendRequest(endpoint)
    }

    fun daysToXats(amount: Int): Map<String, Any> {
        val endpoint = "/dx/$amount"
        return sendRequest(endpoint)
    }

    fun xatsToDays(amount: Int): Map<String, Any> {
        val endpoint = "/x2d/$amount"
        return sendRequest(endpoint)
    }

    fun verifyBanner(url: String): Map<String, Any> {
        val endpoint = "/verifybanner/$url"
        return sendRequest(endpoint)
    }

    fun userGifts(userOrId: String): Map<String, Any> {
        val endpoint = "/usergifts/$userOrId"
        return sendRequest(endpoint)
    }

    fun jinxList(powername: String? = null): Map<String, Any> {
        val endpoint = if (powername != null) {
            "/jinxlist/$powername"
        } else {
            "/jinxlist"
        }
        return sendRequest(endpoint)
    }
}
